### DISCUSSION TITLE
Page data request blocked by adblockers, crashes site

### DISCUSSION THREAD

**Discussion by alexhollenbeck** [gatsbyjs/gatsby#35944] [Help]
Hi everyone,

I'm having a problem on my Gatsby site where links from our ad campaigns are causing the site to crash when the user is using certain adblock software (specifically, AdBlockPlus). What happens is:

1. The user clicks a link from one of our ads, e.g. https://flatfile.com/blog/what-is-a-csv-file-guide-to-uses-and-benefits/?campaignid=15403546517&adgroupid=129480820206
2. Gatsby loads, and then issues a page-data request that looks like: https://flatfile.com/page-data/blog/what-is-a-csv-file-guide-to-uses-and-benefits/page-data.json?campaignid=15403546517&adgroupid=129480820206
3. AdBlockPlus catches the request and blocks it because it contains the string `adgroupid`
4. Without the page data, the site does not render (blank screen).

Obviously we can't have our site crashing for adblock users, so I created a workaround for this. In `gatsby-browser.js`, I look for the `adgroupid` param and strip it (i.e. redirect to the same url without that param) before loading the site:

```exports.onClientEntry = () => {
  window.onload = () => {
    // If adblocker is on, remove adgroupid param to prevent it from blocking
    // page load
    const urlSearchParams = new URLSearchParams(window.location.search)
    const params = Object.fromEntries(urlSearchParams.entries())
    if ('adgroupid' in params && typeof window.__adroll == 'undefined') {
      const newURL = removeURLParameter(window.location.href, 'adgroupid')
      window.location.replace(newURL)
    }
  }
}
```

Unfortunately, this causes us to lose the referrer information for the original click, which we need for reporting on our campaigns.

Ideally, I would like to have some way of intercepting and modifying the page data request (in order to strip adgroupid), instead of having to do a redirect. However I can't find a way to do that in Gatsby. Does anyone know 1. if it's possible to modify page data requests or 2. if there's some other way to prevent the `adgroupid` param from appearing in the request?

EDIT: here is the line of Gatsby code where the query parameters are appended onto the page-data.json request: https://github.com/gatsbyjs/gatsby/blob/e7136908e238d1bc6c3d66c306533c81394535ea/packages/gatsby/cache-dir/loader.js#L27

Ideally I would like to be able to change that so that it does not append query parameters.

**Comment by rasmuswoelk** [comment]
I am having a similar issue where page slugs containing "ads" will fail to load properly and it's really frustrating. I fixed it previously by doing a patch against Gatsby (and related packages) where I would replace the word "ad" with some other word for all page-data files/directories. However, this hacky approach stopped working recently so I had to remove it entirely...

I really wish there was a solution to this because I think we are frustrating and loosing a lot of visitors of our website because of this.
